### ISSUE TITLE
ErrorException fclose(): supplied resource is not a valid stream resource

### ISSUE THREAD

**Issue by piryguiry** [spatie/laravel-medialibrary#1547]
Package version 7.10.1

After do a composer update on my project i have this issue, this occurs when i try to download a zip file of a collection of images, in previous version  before update it worked fine

i do something like

```
public function __invoke(Request $request, Mission $mission): MediaStream
{
    $files = $mission->getMedia('images');
    
    return MediaStream::create($mission->name.'.zip')->addMedia($files);
}

```

And i got this

![Screenshot from 2019-08-30 14-13-05](https://user-images.githubusercontent.com/1696665/64043520-ab0bee80-cb32-11e9-8e80-672bee009628.png)
![Screenshot from 2019-08-30 14-26-09](https://user-images.githubusercontent.com/1696665/64043527-b19a6600-cb32-11e9-8ad3-9491b404ab16.png)

**Comment by freekmurze** [comment]
@rubenvanassche could you take a look?

**Comment by piryguiry** [comment]
This error still persist on version 7.12.0

**Comment by m-rufflesmcghie** [comment]
New to all this coding stuff but after a quick Google it looks like
Their is nothing open for fclose() to close. 
If I'm wrong then sorry just trying to get started with opensource dev :-)

**Comment by rubenvanassche** [comment]
Hi @piryguiry I guess @m-rufflesmcghie is right, there is nothing in the stream. So I think you should check if the files you're trying to put in the zip archive still exist. You could also try to change line 71 from `/vendor/spatie/laravel-medialibrary/src/MediaStream.php`to something like this:

```php
if($stream ! == null){
    fclose($stream);
}
```

If that solves the problem it means the stream was not created or created correctly.

**Comment by piryguiry** [comment]
@rubenvanassche I been following the recommendation, i check the files that i trying to put in the zip, all good at this part.

![Screenshot from 2019-09-06 10-31-59](https://user-images.githubusercontent.com/1696665/64436390-3f75d400-d092-11e9-830c-a02208d420db.png)

I change the line that you're mention, i was did something similar before. but i get the same error.

![Screenshot from 2019-09-06 10-34-31](https://user-images.githubusercontent.com/1696665/64436546-94b1e580-d092-11e9-8c67-b8c7740ed61f.png)

**Comment by rnbroggi** [comment]
Having the exact same problem when trying to download a zip file. 
If it helps, for some reason it seems to work fine on my production app with php 7.1 but not on the one with php 7.3
![screenshot](https://user-images.githubusercontent.com/29467666/64620865-094e9200-d3bb-11e9-83a9-83253ff0f7f0.PNG)

**Comment by piryguiry** [comment]
@rubenvanassche, @rnbroggi Yes it's only the zip download fails, in my case on production runs fine with php 7.3.9, and the package version 7.6.0, but after update have this issue.

**Comment by rnbroggi** [comment]
@piryguiry Yup, switched back to V7.6.0 and works. Looks like there's indeed an issue with the new version

**Comment by bgarrison25** [comment]
Same here. I just rolled back from 7.9 -> 7.6 and it started working again. Had the same issue.

**Comment by ziming** [comment]
7.6.3 works for me, after that the newer versions has this issue. I'm using PHP 7.3 if it helps

**Comment by ncatanchin** [comment]
I ran into both of these problems today.

The first was "solved" by checking `is_resource` before attempting `fclose()`

```php
                if (is_resource($stream)) {
                    fclose($stream);
                }
```

The second problem is related to Zipstream not sending any HTTP headers by default, which leads to the binary zip data being shown directly in browser:

```
            $options = new \ZipStream\Option\Archive();
            $options->setContentType('application/octet-stream');
            $options->setComment('test zip file.');
            // Send http headers (default is false)
            $options->setSendHttpHeaders(true);
            // HTTP Content-Disposition.  Defaults to 'attachment', where FILENAME is the specified filename. Note that this does nothing if you are not sending HTTP headers.
            $options->setContentDisposition('attachment');

            // Set the content type (does nothing if you are not sending HTTP headers)
            $options->setContentType('application/x-zip');
```

The problems would appear to not be related to `laravel-medialibrary` specifically.

References:
* https://github.com/maennchen/ZipStream-PHP/wiki/Available-options
* https://github.com/maennchen/ZipStream-PHP/blob/master/CHANGELOG.md

**Comment by rubenvanassche** [comment]
Hi, guys I was on vacation but looking into this today!

**Comment by freekmurze** [comment]
The fix by @rubenvanassche is now merged and tagged. This problem is fixed in v7.12.3 of the package.
